### DISCUSSION TITLE
fatal error: 'bpf/bpf_helpers.h' file not found

### DISCUSSION THREAD

**Discussion by karim-manaouil** [iovisor/bcc#4584] [Q&A]
Hi,

I installed `bcc` via the Debian's package manager. I tested it and it works in a very simple scenario with only `bpf_printk` in a syscall.

However, when I included `bpf/bpf_helpers.h` to have the definition of `BPF_SNPRINTF`, I got the error in the title.

I am not sure how `bcc` finds its headers. Assuming the usual way on Linux, I installed `libbpf-dev`, as well Linux kernel headers, and I have `/usr/include/bpf/bpf_helpers.h` on my installation. So, I don't really know why I'm getting that error message. Would appreciate your help!

Cheers!

**Comment by chenhengqi** [comment]
You should use `bpf_snprintf` instead.

**Comment by hatamiarash7** [comment]
I have this problem too. It's a simple Python program.

```
from bcc import BPF

prog = """
#include <bpf/bpf_helpers.h>

int hello(void *ctx) {
    bpf_trace_printk("Hello, World!\\n");
    return 0;
}
"""

b = BPF(
    text=prog,
    cflags=[
        "-I/usr/include/bpf",
        "-I/usr/include/x86_64-linux-gnu",
        "-I/usr/include/linux",
    ],
    debug=1,
)
b.attach_kprobe(event=b.get_syscall_fnname("clone"), fn_name="hello")
```

This will give me an error:

```
/virtual/main.c:2:10: fatal error: 'bpf/bpf_helpers.h' file not found
```

I'm sure that everything is installed. Even if I used `cflags` to pass the path manually, it did not work.

**Comment by MisuStefanLeonard** [comment]
Same problem as @hatamiarash7 . Any news ?

**Comment by embedded4ever** [comment]
Anyone faced with the issue, could try : 

Ubuntu :
`apt install libbpf-dev`

Fedore :
`sudo dnf install libbpf-devel`
